includes/Core/Modules/Modules.php

```diff
@@ -1131,14 +1131,14 @@ function( Module $module ) {
 	 *
 	 * @since 1.50.0
 	 *
-	 * @return array List of recoverable modules.
+	 * @return Module_With_Owner[] array List of recoverable modules.
```

Comment:
@hussain-t this return type was intended for the `get_shareable_modules` method – this method currently returns a list of strings rather than module objects.

This wasn't spelled out in the AC or IB but `get_*_modules` methods on this class return an array of `Module` instances, but this one currently returns a list of slugs.

Let's update this method to return a filtered list of `Module` objects instead (like `get_shareable_modules` does only with the logic we have here) in which case this return type can stay.

Then we can `wp_list_pluck` the slugs from the list of modules above for the dashboard sharing data.